airflow/www/static/js/grid/details/content/taskInstance/Logs/index.jsx

```diff
@@ -0,0 +1,163 @@
+
+import React, { useState } from 'react';
+import {
+  Text,
+  Box,
+  Flex,
+  Divider,
+  Textarea,
+  Button,
+  Checkbox,
+} from '@chakra-ui/react';
+
+import { getMetaValue } from '../../../../../utils';
+import LogLink from './LogLink';
+import useTaskLog from '../../../../api/useTaskLog';
+import LinkButton from '../../../../components/LinkButton';
+
+const showExternalLogRedirect = getMetaValue('show_external_log_redirect') === 'True';
+const externalLogName = getMetaValue('external_log_name');
+const logUrl = getMetaValue('log_url');
+
+const getLinkIndexes = (tryNumber) => {
+  const internalIndexes = [];
+  const externalIndexes = [];
+
+  [...Array(tryNumber + 1 || 0)].forEach((_, index) => {
+    if (index === 0 && tryNumber < 2) return;
+    const isExternal = index !== 0 && showExternalLogRedirect;
+    if (isExternal) {
+      externalIndexes.push(index);
+    } else {
+      internalIndexes.push(index);
+    }
+  });
+
+  return [internalIndexes, externalIndexes];
+};
+
+const Logs = ({
+  dagId,
+  dagRunId,
+  taskId,
+  executionDate,
+  tryNumber,
+  isGroup,
+}) => {
+  const [internalIndexes, externalIndexes] = getLinkIndexes(tryNumber);
+  const [selectedAttempt, setSelectedAttempt] = useState(1);
+  const [shouldRequestFullContent, setShouldRequestFullContent] = useState(false);
+  const { data, isSuccess } = useTaskLog({
+    dagId,
+    dagRunId,
+    taskId,
+    taskTryNumber: selectedAttempt,
+    fullContent: shouldRequestFullContent,
+    enabled: (!isGroup),
+  });
+
+  const params = new URLSearchParams({
+    task_id: taskId,
+    execution_date: executionDate,
+  }).toString();
+
+  return (
+    <>
+      {tryNumber > 0 && (
+      <>
+        <Text as="span"> (by attempts)</Text>
+        <Box>
+          <Flex my={1} justifyContent="space-between">
+            <Flex flexWrap="wrap">
+              {internalIndexes.map((index) => (
+                <Button
+                  key={index}
+                  variant="ghost"
+                  colorScheme="blue"
+                  onClick={() => setSelectedAttempt(index)}
+                >
+                  {index}
+                </Button>
+              ))}
+            </Flex>
+            <Flex>
+              <Checkbox
+                onChange={() => setShouldRequestFullContent((previousState) => !previousState)}
+                px={4}
+              >
+                <Text as="strong">Full Logs</Text>
+              </Checkbox>
+              <LogLink
+                index={selectedAttempt}
+                dagId={dagId}
+                taskId={taskId}
+                executionDate={executionDate}
+                isInternal
+              />
+              <LinkButton
+                href={`${logUrl}&${params}`}
+              >
+                See More
+              </LinkButton>
+            </Flex>
+          </Flex>
+        </Box>
+        {
+          isSuccess && (
+          <Textarea readOnly defaultValue={data} height={350} />
```

Comment:
Let's make sure we scroll to the latest logs (bottom of the textarea)

Reply:
Also, let's make it a `<Code />` block to visually separate it from other text

Reply:
Good Idea, done both of them :smile: